File: mvp-workflow-app/backend/models/workflow.py
from pydantic import BaseModel
from typing import List, Literal

class Input(BaseModel):
    name: str
    type: str

class Output(BaseModel):
    name: str
    type: str

class Step(BaseModel):
    name: str
    description: str
    command: str
    inputs: List[Input]
    outputs: List[Output]
    dependencies: List[str]
    status: Literal['保留中', '実行中', '完了', '失敗']
    processing_time: Literal['短', '中', '長']

class Workflow(BaseModel):
    name: str
    description: str
    steps: List[Step]

File: mvp-workflow-app/backend/src/models/workflow.ts
export interface Input {
  name: string;
  type: string;
}

export interface Output {
  name: string;
  type: string;
}

export interface Step {
  name: string;
  description: string;
  command: string;
  inputs: Input[];
  outputs: Output[];
  dependencies: string[];
  status: "保留中" | "実行中" | "完了" | "失敗";
  processing_time: "短" | "中" | "長";
}

export interface Workflow {
  name: string;
  description: string;
  steps: Step[];
}
